[make] Generate haxelib binary with `nekotools boot -c` on Mac/Linux

tobil4sk opened the pull request:
`nekotools boot` is a bit of a hack because it simply copies the neko exectuable and injects in the neko bytecode. This can cause issues with some tools that deal with executables on Linux/Mac. https://github.com/HaxeFoundation/neko/issues/130#issuecomment-218610395

This started happening on Arch linux, where the bytecode somehow gets stripped away during packaging so haxelib becomes unusable because it is reverted to the regular neko binary. See: #11652.

`nekotools boot -c` will instead generate a .c file which can then be compiled manually. I've changed this for Mac and Linux, but for Windows it will continue to use `nekotools boot` as it is simpler and doesn't cause issues there as far as I'm aware.

I'm not sure if this may have packaging implications now that a c compiler is used directly in the make file?

tobil4sk committed:
[make] Generate haxelib binary with `nekotools boot -c` on Mac/Linux

`nekotools boot` is a bit of a hack and some tools that deal with executables on linux can strip away the haxelib bytecode if it is done this way.

`nekotools boot -c` will generate a .c file which can then be compiled manually.

tobil4sk committed:
[make] Fix for unix platforms if PLATFORM is defined

tobil4sk committed:
[make] Set the HAXE_STD_PATH while building haxelib

Otherwise it doesn't work if haxe isn't already installed, or it might use the wrong version of the standard library

tobil4sk committed:
[ci] Install missing neko headers during setup

They are now needed for compiling haxelib

tobil4sk committed:
[ci] Ensure /usr/local/include exists before installing neko

RMSPORTS1902 commented:
![vini jr🤍](https://github.com/HaxeFoundation/haxe/assets/157073946/554f319d-8d89-4585-b4f7-670ac874cd44)

RMSPORTS1902 commented:
come on

tobil4sk committed:
[ci] Install neko headers for earthly builds as well

tobil4sk committed:
[make] Omit HAXE_STD_PATH when building haxelib on windows

For some reason this environment variable isn't working here on Windows

tobil4sk committed:
[ci] Continue building x64 haxelib on mac arm64 for now

We don't have neko binaries for mac arm64 yet

`nekotools boot run.n` was previously also producing an x64 binary

tobil4sk committed:
[make] Set rpath on mac when building haxelib

It looks like /usr/local/lib is no longer used as a default path in some cases.

tobil4sk commented:
Since we are building the executable on macos-13 (rather than macos-12) `/usr/local/lib` is no longer a default loader path for the executable, so we need to add it as an rpath. The neko binary is built with macos-12 (and previously 11) which is why it wasn't necessary when generating the haxelib executable from the neko executable.

Simn commented:
Should I merge this or do we need to do something else first?

tobil4sk committed:
[make] Set std path for haxelib build on windows

tobil4sk commented:
> Should I merge this or do we need to do something else first?

I don't think there's anything left to do here.
